packages/cli/lib/projects.js

```diff
@@ -618,6 +620,30 @@ const logFeedbackMessage = buildId => {
   }
 };
 
+const createProjectComponent = async (component, name) => {
+  let componentName = name;
+
+  if (component.type === PROJECT_COMPONENT_TYPES.APP_FUNCTION) {
```

Comment:
I would vote to force the repo config to specify the extension. That way the CLI never needs to know anything about the type of asset that's being created. Ideally we can keep the CLI as "dumb" as possible here so teams can self-serve by adding new components and other assets to the `hubspot-project-components` repo without having to make any changes here in the CLI.